src/Tinfoil/Data/Signing.hs

```diff
@@ -31,30 +34,26 @@ parseSignatureVersion :: Text -> Maybe SignatureVersion
 parseSignatureVersion "v1" = pure SignatureV1
 parseSignatureVersion _ = Nothing
 
--- | Keyed-hash algorithm designator, for inclusion as a request parameter.
-data KeyedHashFunction =
-    HMAC_SHA256
-  deriving (Eq, Show, Enum, Bounded, Generic)
-
-instance NFData KeyedHashFunction
-
-renderKeyedHashFunction :: KeyedHashFunction -> Text
-renderKeyedHashFunction HMAC_SHA256 = "HMAC-SHA256"
-
-parseKeyedHashFunction :: Text -> Maybe KeyedHashFunction
-parseKeyedHashFunction "HMAC-SHA256" = pure HMAC_SHA256
-parseKeyedHashFunction _ = Nothing
-
-newtype SigningKey =
-  SigningKey {
-    unSigningKey :: ByteString
+newtype KeyId =
+  KeyId {
+    unKeyId :: ByteString
   } deriving (Eq, Show, Generic)
 
-instance NFData SigningKey
+instance NFData KeyId where rnf = genericRnf
 
 newtype RequestScope =
   RequestScope {
     unRequestScope :: ByteString
   } deriving (Eq, Show, Generic)
 
-instance NFData RequestScope
+instance NFData RequestScope where rnf = genericRnf
+
+newtype Signature =
+  Signature {
+    unSignature :: MAC
```

Comment:
Should be fixed in the latest commit, I just removed the `Signature` type for now as nothing's using it yet now that `macBytes` returns a `MAC`. Name bikeshedding all good, agree it's worth getting the terminology right from the start.

I think probably reserving "Signing" to refer to asymmetric in keeping with typical crypto usage isn't a bad idea, and "Authenticating" or just "MAC" for symmetric?